lib/util/identifier.js

```diff
@@ -259,13 +301,15 @@ const _absolutify = (context, request) => {
 		.join("!");
 };
 
-const absolutify = makeCacheable(_absolutify);
+const absolutify = makeCacheableWithContext(_absolutify);
 exports.absolutify = absolutify;
 
 const PATH_QUERY_FRAGMENT_REGEXP =
 	/^((?:\0.|[^?#\0])*)(\?(?:\0.|[^#\0])*)?(#.*)?$/;
+const PATH_QUERY_REGEXP = /^((?:\0.|[^?#\0])*)(\?.*)?$/;
```

Comment:
```suggestion
const PATH_QUERY_REGEXP = /^((?:\0.|[^?\0])*)(\?.*)?$/;
```

This should not stop at `#`

Reply:
yeah, right now all works because of escaping..